src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionHelper.cs

```diff
@@ -178,9 +190,18 @@ public static bool IsSingleArgumentIndexer(Expression expression)
             }
 
             // Find default property (the indexer) and confirm its getter is the method in this expression.
-            return declaringType.GetRuntimeProperties().Any(
-                property => (string.Equals(defaultMember.MemberName, property.Name, StringComparison.Ordinal) &&
-                    property.GetMethod == methodExpression.Method));
+            var runtimeProperties = declaringType.GetRuntimeProperties();
+            for (int i=0; i< runtimeProperties.Count(); i++ )
```

Comment:
Here the trade off is allocation versus computational cost. The ToArray() adds an allocation whereas the original one has more instructions. In my opinion, I would prefer the later one because in most of the cases, we break early as we find the matching property,